server/channels/app/channel.go

```diff
@@ -3632,7 +3633,74 @@ func (a *App) ConvertGroupMessageToChannel(c request.CTX, convertedByUserId stri
 	_ = a.postMessageForConvertGroupMessageToChannel(c, gmConversionRequest.ChannelID, convertedByUserId, users)
 
 	// the user conversion the GM becomes the channel admin.
-	_, appErr = a.UpdateChannelMemberSchemeRoles(c, gmConversionRequest.ChannelID, convertedByUserId, false, true, true)
+	if convertedByUserId != "" {
+		_, appErr = a.UpdateChannelMemberSchemeRoles(c, gmConversionRequest.ChannelID, convertedByUserId, false, true, true)
+		if appErr != nil {
+			return nil, appErr
+		}
+	} else {
+		// if there is no one converting the GM, then the everyone should become the channel admin.
+		for _, user := range users {
+			_, appErr = a.UpdateChannelMemberSchemeRoles(c, gmConversionRequest.ChannelID, user.Id, false, true, true)
+			if appErr != nil {
+				return nil, appErr
+			}
+		}
+	}
+
+	return updatedChannel, nil
+}
+
+func (a *App) convertDirectMessageToChannel(c request.CTX, leavingUsername string, channelID string) (*model.Channel, *model.AppError) {
```

Comment:
This needs an app level test. See `TestConvertGroupMessageToChannel` for reference.